metaflow/plugins/cards/card_datastore.py

```diff
@@ -102,16 +116,15 @@ def _make_path(self, base_pth, pathspec=None):
                 task_id,
                 "cards",
             ]
-
         if sysroot == "" or sysroot == None:
             pth_arr.pop(0)
         return os.path.join(*pth_arr)
 
-    def _get_card_path(self, base_pth=""):
-        return self._make_path(
-            base_pth,
-            pathspec=self._pathspec,
-        )
+    def _get_write_path(self, base_pth=""):
```

Comment:
To ensure that this is not a breaking change right away, we can write the cards to both the locations (with and without step name) and deprecate the second write path (without step name) 3 months from now. The reads still look for the path with the step name first.

Reply:
If we don't do this, the UI will start breaking almost immediately for all deployments.

Reply:
That would work. The UI would indeed not show cards until updated. Writing to both locations would solve this and doesn't ahve a huge cost. Maybe protect it with a flag though (hidden). That way I can disable it because I don't really need it (and would update the UI right away).